aspnetcore/testing/razor-pages-testing.md

```diff
@@ -0,0 +1,238 @@
+---
+title: Razor Pages unit and integration testing in ASP.NET Core
+author: guardrex
+description: Learn how to create unit and integration tests for Razor Pages apps.
+ms.author: riande
+manager: wpickett
+ms.date: 11/20/2017
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: testing/razor-pages-testing
+---
+# Razor Pages unit and integration testing in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+ASP.NET Core supports unit and integration testing of Razor Pages apps. Testing the data access layer (DAL), page models, and integrated page components helps ensure:
+
+* Parts of a Razor Pages app work independently and together as a unit during app construction.
+* Classes and methods have limited scopes of responsibility.
+* Additional documentation exists on how the app should behave.
+* Regressions, which are errors brought about by updates to the code, are found during automated building and deployment.
+
+This topic assumes that you have a basic understanding of Razor Pages apps, unit testing, and integration testing. If you're unfamiliar with Razor Pages apps or testing concepts, see the following topics:
+
+* [Introduction to Razor Pages](xref:mvc/razor-pages/index)
+* [Getting started with Razor Pages](xref:tutorials/razor-pages/razor-pages-start)
+* [Unit testing C# in .NET Core using dotnet test and xUnit](/dotnet/articles/core/testing/unit-testing-with-dotnet-test)
+* [Integration testing](xref:testing/integration-testing)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/testing/razor-pages-testing/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The sample project is composed of two apps:
+
+| App         | Project folder                        | Description |
+| ----------- | ------------------------------------- | ----------- |
+| Message app | *src/RazorPagesTestingSample*         | Allows a user to add, delete, delete all, and analyze messages. |
```

Comment:
We should probably be more specific with what's meant by "delete", since "delete all" is also listed. Should this say something like "delete one"?